src/librustc_codegen_ssa/mir/mod.rs

```diff
@@ -559,9 +581,35 @@ fn arg_local_refs<'a, 'tcx: 'a, Bx: BuilderMethods<'a, 'tcx>>(
             indirect_operand.store(bx, tmp);
             tmp
         } else {
-            let tmp = PlaceRef::alloca(bx, arg.layout, &name);
-            bx.store_fn_arg(arg, &mut llarg_idx, tmp);
-            tmp
+            if fx.fn_ty.variadic && last_arg_idx.map(|idx| arg_index == idx).unwrap_or(false) {
+                let va_list_impl = match arg_decl.ty.ty_adt_def() {
+                    Some(adt) => adt.non_enum_variant(),
+                    None => bug!("va_list language item improperly constructed")
+                };
+                match tcx.type_of(va_list_impl.fields[0].did).sty {
+                    ty::Ref(_, ty, _) => {
+                        // If the underlying structure the `VaList` contains is a structure,
+                        // we need to allocate it (e.g. X86_64 on Linux).
```

Comment:
I think we decided on commas after `e.g.` in the codebase now (that's how I'm normalising it)...

```
// If the underlying structure the `VaList` contains is a structure,
// we need to allocate it (e.g., x86_64 on Linux).
```

So, this explains what you're doing above in `va_start`/`va_end` too. You can either explain it briefly in both places (briefer in one place?), or just refer from one place in code to the other place.

Reply:
I'll add a doc comment to the trait definition.